src/xdocs/config_misc.xml

```diff
@@ -69,6 +88,23 @@
   &lt;property name=&quot;javaStyle&quot; value=&quot;false&quot;/&gt;
 &lt;/module&gt;
         </source>
+        <p>
+          Example:
+        </p>
+        <source>
+public class MyClass {
+  int[] nums; // violation
+  String strings[]; // OK
+
+  char[] toCharArray() { // OK
+    return null;
+  }
+
+  byte getData()[] { // violation (only Java style allowed)
```

Comment:
I think we need to be consistent and allow that weird location of `[]` when property is set to false.

In my previous comment I meant that I do not know who likes `String strings[];`, even 20 year ago, it was not good  way to declare array.

Reply:
Ok, sounds good.

Should `javaStyle` = `false` mean all array return types for methods have to be like `byte getData()[]`? Or should the be a separate property for methods?

On the one hand, I wonder how many people actually set `javaStyle` to `false`, so maybe it's not such a big deal either way. But it seems like this weird `byte getData()[]` syntax isn't really "C style" since functions/methods in C can't actually return `[]` types.

Maybe add a `tokens` property that can include `METHOD_DEF`, or something specific like `ignoreMethodReturnTypes`?

Then a user could hypothetically enable a "C style" check but still choose to enforce "Java style" method return types.

Reply:
Sounds good. So should we say: method return types are only checked when `javaStyle` is `true`? Then if that property is `false`, either style is OK?

Reply:
@romani Please confirm.
@esilkensen If logic is changing, documentation also needs to be updated.

Reply:
@rnveach I tried to update the documentation to reflect the current logic with my last commit. Is there a spot in particular you think it should be changed?

Anyway, thank you both for your patience - I am sorry for the churn, I just want to implement the logic that you would like to see for this check.

* Initial version of the PR: method return type always checked (and must be "Java style")
* Current version of the PR: method return type only checked when `javaStyle` is `true`

I was interpreting your first comments @romani as agreeing with that first version, then interpreted your most recent comment as saying we should have the second version.

Any approach is ok with me (or maybe a third one with a separate property configuring the method return check specifically?) - just let me know what you would like and I will make the changes.

Reply:
varieble/field type and method type  are always checked.
if javaStyle="true" (default), then "[]" should be on type, only.
if javaStyle="false" then "[]" can be at other place.

Please extend documentation with examples to clearly show user what behavior will be. Examples works better than some explanation of logic.

Reply:
Sorry, one more clarification:
>varieble/field type and method type are always checked.
if javaStyle="true" (default), then "[]" should be on type, only.
if javaStyle="false" then "[]" can be at other place.

With the current logic in the PR: if javaStyle="false", then "[]" _can_ be at the other place (for method return types), but it doesn't _have_ to be.

Do you want it to be: if javaStyle="false" then "[]" should be at other place, only?

The reason I thought it might make sense to not make `char[] toCharArray()` a violation when javaStyle="false" is that it isn't really "C style" since it is not possible to actually return a `[]` type in C/C++ -- I don't know where this `char toCharArray()[]` syntax came from, but it doesn't seem like C/C++.

I thought it might be likely that a user would want to enforce `int nums[];` style while still writing `char[] toCharArray()` for methods - but, I wonder how many users actually enable this check with javaStyle="false" in the first place - maybe it's no big deal either way.

Anyway, I am happy to make the change so that javaStyle="false" means `char[] toCharArray()` is a violation, and will do that now.

Reply:
Just rechecked, as a long time passed from C time...
tested at https://www.onlinegdb.com/online_c_compiler
```
#include <stdio.h>

int fun()[] 
{
    static int arr[100];    
    return arr; 
} 

int main()
{
    int ptr[] = fun(arr); 
    return 0;
}
```
error:
```
main.c:4:5: error: ‘fun’ declared as function returning an array
 int fun()[] 
     ^
main.c: In function ‘fun’:
main.c:7:12: warning: return makes integer from pointer without a cast [-Wint-conversion]
     return arr; 
            ^
main.c: In function ‘main’:
main.c:12:21: error: ‘arr’ undeclared (first use in this function)
     int ptr[] = fun(arr); 
                     ^
main.c:12:21: note: each undeclared identifier is reported only once for each function it appears in
```

You are right.
it is not possible(compilation error) to return array as value from function.

https://stackoverflow.com/a/14297191
https://stackoverflow.com/questions/3473438/return-array-in-a-function

So it is ok to forbid `char toCharArray()[]` when `javaStyle="false"`, but we need to explain this in documentation, as it is very tiny nuance.

@rnveach , please confirm that you are ok with such behavior.

Reply:
@esilkensen @romani I am ok. It sounds like we are reverting back to the first instance of this PR where all `[]`s on method name are forbidden.

Reply:
Thanks @rnveach , @romani, I have updated the code and documentation with an extra note - let me know any further changes you would like to see. (Also, next time I will ask questions/clarify behavior in the issue first, if unsure about the implementation, hopefully for a smoother PR!)